### ISSUE TITLE
Fetch Range of code

### ISSUE THREAD

**Issue by TomPeirs** [DavidWells/markdown-magic#46]
Hi, I filtered the open and closed issues and did not come across this feature request that I am curious about.
I am currently getting code from URL and put in markdown with markdown-magic.

Idea:
We have big files of code in our public git repository, and I do not want to create gists from each particular method or function of interest  (because I want to avoid to maintain multiple files) in order to show only a partial code snippet in markdown.

I would not know directly how to use the custom transforms but it would be cool to have the ability the define a range with Lines of Code which will then be replaced with code from the file.

Are there any implementations where this is done already?

Kind regards,
Tom

**Comment by forresst** [comment]
To my knowledge, there is currently no solution for your request.

One or more additional options in the [`CODE' transform](https://github.com/DavidWells/markdown-magic#code) would have to be provided to manage this request

**Comment by DavidWells** [comment]
Adding options to `CODE` would work. Something like `lines`

```
<!-- AUTO-GENERATED-CONTENT:START (CODE:src=./relative/path/to/code.js&lines=22-44) -->
This content will be dynamically replaced with code from the file lines 22 through 44
<!-- AUTO-GENERATED-CONTENT:END -->
```

Happy to merge in a PR for this if someone writes it 😃

**Comment by TomPeirs** [comment]
I would very much like that 😁😁

**Comment by forresst** [comment]
I will take care of this option as soon as possible.

**Comment by TomPeirs** [comment]
What could also be even more stable is to fetch code blocks not just on lines of code but based of a code comment similar to:

// code


Because sometimes as the external file changes you might have to change the Lines of Code in your markdown as well.

**Comment by forresst** [comment]
@TomPeirs The functionality to retrieve a piece of code by specifying lines is available in the new version of markdown-magic 2.0.0.

The idea of having an identifier to fetch a piece of code should be discussed again. I advise you to create a new issue
